test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java

```diff
@@ -231,9 +235,15 @@ public void interceptBeforeAllMethod(Invocation<Void> invocation, ReflectiveInvo
         }
         T result;
         ClassLoader old = Thread.currentThread().getContextClassLoader();
+        Class<?> requiredTestClass = extensionContext.getRequiredTestClass();
         try {
-            Thread.currentThread().setContextClassLoader(extensionContext.getRequiredTestClass().getClassLoader());
+            Thread.currentThread().setContextClassLoader(requiredTestClass.getClassLoader());
             result = invocation.proceed();
+        } catch (NullPointerException e) {
+            System.err.println(
+                    "When using constructor injection in a test, the only legal operation is to assign the constructor values to fields. Offending class is "
```

Comment:
I would wrap the exception with this, not log it, so it shows up in the test failures.

Reply:
Thanks for the suggestion, it looks much much better in the logs!